packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/components/TableBody.js

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { IconButton, Tbody, Text, Tr, Td, Row } from '@strapi/parts';
+import { EditIcon, DeleteIcon } from '@strapi/icons';
+import { CheckPermissions } from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
+
+import { getTrad } from '../../../../utils';
+import pluginId from '../../../../pluginId';
+
+const TableBody = ({
+  sortedRoles,
+  canDelete,
+  permissions,
+  setRoleToDelete,
+  setShowConfirmDelete,
+  showConfirmDelete,
+}) => {
+  const { formatMessage } = useIntl();
+  const { push } = useHistory();
+
+  const checkCanDeleteRole = useCallback(
```

Comment:
What is the perf implication that forced you to rely on this hook?

Reply:
Good question 🤔 .  This was an existing function I left in place.  I'm not sure I completely understand what the performance benefit would be here other than that `.includes()` is going to be called in a loop.  If the list is long and `canDelete` hasn't changed maybe it is better to use the same instance of the function?  By removing it nothing jumped out at me in the react dev tools profiler.  Maybe @soupette has further insight?

Reply:
You can remove it I think the old implementation used `useCallback` because a lot of function where in other's dependencies array so at some point we had to use it for every function in order to prevent entering in an infinite loop.
You can get rid of all the useCallback hooks in the list view.